content/influxdb/clustered/query-data/troubleshoot-and-optimize/report-query-performance-issues.md

```diff
@@ -310,3 +323,71 @@ curl --get "https://{{< influxdb/host >}}/query" \
 {{< /code-tabs-wrapper >}}
 
 {{% /code-placeholders %}}
+
+### Gather system information
+
+{{% warn %}}
+#### May impact cluster performance
+
+Querying InfluxDB v3 system tables may impact write and query
+performance of your {{< product-name omit=" Clustered" >}} cluster.
+Use filters to [optimize queries to reduce impact to your cluster](/influxdb/clustered/admin/query-system-data/#optimize-queries-to-reduce-impact-to-your-cluster).
+
+<!--------------- UPDATE THE DATE BELOW AS EXAMPLES ARE UPDATED --------------->
+
+#### System tables are subject to change
+
+System tables are not part of InfluxDB's stable API and may change with new releases.
+The provided schema information and query examples are valid as of **September 20, 2024**.
+If you detect a schema change or a non-functioning query example, please
+[submit an issue](https://github.com/influxdata/docs-v2/issues/new/choose).
+
+<!--------------- UPDATE THE DATE ABOVE AS EXAMPLES ARE UPDATED --------------->
+{{% /warn %}}
+
+If queries are slow for a specific table, run the following system queries to collect information for troubleshooting.
+
+- [Collect table information](#collect-table-information)
+- [Collect compaction information for the table](#collect-compaction-information-for-the-table)
+- [Collect partition information for multiple tables](#collect-partition-information-for-multiple-tables)
+
+In your queries, replace the following:
+
+- {{% code-placeholder-key %}}`TABLE_NAME`{{% /code-placeholder-key %}}: the
+  table to retrieve information about 
+
+#### Collect table information
+
+{{% code-placeholders "TABLE_NAME" %}}
+```sql
+SELECT *
+FROM system.tables
+WHERE table_name = 'TABLE_NAME';
+```
+{{% /code-placeholders%}}
+
+#### Collect compaction information for the table
+
+{{% code-placeholders "TABLE_NAME" %}}
+```sql
+SELECT *
+FROM system.compactor
+WHERE table_name = 'TABLE_NAME';
```

Comment:
If I understand correctly, if we add `and partition_key = '<your_partition_key>'` , it will be even faster if you only want to see that partition. I think we may want to add another example to add this filter.

You may want to run the query and see if there is `partition_key` in the system.compactor` table. Do the same for other system tables. The point is we want to provide the cheapest query possible

Reply:
And @appletreeisyellow built these features and knows its the most

Reply:
Yes, combining `table_name` with `partition_key` and `partition_id` will be even faster. 

You can add these two examples:

```sql
SELECT * 
FROM system.compactor 
WHERE
  table_name = 'TABLE_NAME' 
    AND partition_key = 'PARTITION_KEY';
```

```sql
SELECT * 
FROM system.compactor 
WHERE
  table_name = 'TABLE_NAME' 
    AND partition_id = PARTITION_ID;
```

Probably also show the users how to retrieve `partition_id`: https://docs.influxdata.com/influxdb/clustered/admin/query-system-data/#retrieve-a-partition-id

Reply:
@appletreeisyellow there is already a section describing that optimization just above the section you linked: https://docs.influxdata.com/influxdb/clustered/admin/query-system-data/#filter-by-partition-id

Apologies if you suggesting that the optimization be recommended in both places, you probably already know about the section I linked.